.github/workflows/cicd.yaml

```diff
@@ -20,6 +23,8 @@ jobs:
       - run: true
 
   Static-Check:
+    needs:
+      - Authorize
```

Comment:
@pankajkoti, I believe you're right. Since we don't have particularly expensive tests, perhaps we could just remove the Authorize job altogether, WDYT?

Reply:
yes, probably. we can re-add it back later if we see we added expensive tests.

Reply:
Addressed in 8f2d8085f59a3cbdb8bb29ff09387d4146d4c8af